test/test_response.py

```diff
@@ -198,7 +200,9 @@ def test_io(self):
         self.assertRaises(IOError, resp3.fileno)
 
     def test_io_closed_consistently(self):
-        hlr = httplib.HTTPResponse(socket.socket())
+        sock = socket.socket()
+        self.addCleanup(sock.close)
```

Comment:
I don’t think that works – if I replace these three lines with

```python
hlr = httplib.HTTPResponse(socket.socket())
self.addCleanup(hlr.close)
```

I get two `ResourceWarning`s when I run the tests. If you look at the [source code for HTTPResponse](https://github.com/python/cpython/blob/a632d00a1cbb6f4f8978cb45dd9789658cceb7c8/Lib/http/client.py#L217), it creates a file object associated with the socket, and the file object gets closed when you call `hlr.close()`, but the socket itself is left open.